drivers/flash/flash_stm32h7x.c

```diff
@@ -60,11 +78,25 @@ static inline void _flash_stm32_sem_give(const struct device *dev)
 #define flash_stm32_sem_give(dev)
 #endif
 
-
 bool flash_stm32_valid_range(const struct device *dev, off_t offset,
 			     uint32_t len,
 			     bool write)
 {
+#if defined(DUAL_BANK)
+	if (DISCONTINUOUS_BANKS) {
```

Comment:
Why not using a preprocessor directive : `#if (DISCONTINUOUS_BANK > 0)` ? Could be merged with the previous line

Reply:
This is not possible because REAL_FLASH_SIZE_KB  is computed at runtime by reading Hardware register.